### ISSUE TITLE
CustomerStateRequestProcessor stores the entire customer in the session

### ISSUE THREAD

**Issue by smmckay** [BroadleafCommerce/BroadleafCommerce#1372]
Login event publishing stores the last published event in the session. The event has a reference to the customer. If sessions are stored outside the JVM this is Very Not Good. Otherwise it's just a waste, I suppose. The only thing that gets used is the username, so it seems like that could be stored in the event rather than the whole customer.

**Comment by phillipuniverse** [comment]
Agree this is a bug. I'm not even really sure what the point of putting that event in session at all. As you indicated that seems Very Not Good&trade;. Assigning to myself to review further.

**Comment by phillipuniverse** [comment]
I pushed up a fix for this but now I'm a little confused. We're storing an entire Customer object in session for all anonymous users so we can avoid unnecessary database hits/data. Why wouldn't that also be a problem or would it?

@smmckay what problem was actually manifested? Was it just that maybe the customer instance wasn't hooked up to a Hibernate session?

**Comment by smmckay** [comment]
This came up from using a weird (non-Java-standard) serialization mechanism. Instead it traverses the object graph, so a customer with an order history will pull along basically the entire catalog. However, the issue I ran into was something different…session activation was busted IIRC due to Hibernate’s lazy-loaded collections having trouble finding the SessionFactory after weird deserialization. It may have been something else, this was a while ago. I filed the whole thing under Doing Weird Stuff Breaks Things and didn’t notice this sub-issue until recently.

The anonymous customer is only stored in the session if it has a NullOrderImpl cart, so that customer is small. It also doesn’t have any Hibernate classes in the object graph to interfere with weird serialization or weird deserialization.
